atef/widgets/config/result_summary.py

```diff
@@ -0,0 +1,497 @@
+"""
+Widgets for summarizing the results of a run.
+"""
+from __future__ import annotations
+
+import csv
+import dataclasses
+from typing import Any, List, Set
+
+from qtpy import QtCore, QtGui, QtWidgets
+from qtpy.QtCore import Qt
+
+from atef.config import PreparedFile
+from atef.enums import Severity
+from atef.procedure import PreparedProcedureFile
+from atef.type_hints import AnyDataclass
+from atef.walk import walk_config_file, walk_procedure_file
+from atef.widgets.core import DesignerDisplay
+from atef.widgets.utils import insert_widget
+
+
+@dataclasses.dataclass
+class ResultInfo:
+    """ Normalized, and slightly processed view of configs/steps with results """
+    status: Severity
+    name: str
+    reason: str
+
+    # An un-prepared dataclass, to match the tree views (will not hold the result)
+    origin: AnyDataclass
+
+    @property
+    def type(self):
+        return type(self.origin).__name__
+
+
+class ResultModel(QtCore.QAbstractTableModel):
+    """
+    Item model for results.  Read-Only.
+    To be proxied for searching
+    """
+    result_info: List[ResultInfo]
+    result_icon_map = {
+        # check mark
+        Severity.success: ('\u2713', QtGui.QColor(0, 128, 0, 255)),
+        Severity.warning : ('?', QtGui.QColor(255, 165, 0, 255)),
+        # x mark
+        Severity.internal_error: ('\u2718', QtGui.QColor(255, 0, 0, 255)),
+        Severity.error: ('\u2718', QtGui.QColor(255, 0, 0, 255)),
+        'N/A': ('nothing', QtGui.QColor())
+    }
+
+    def __init__(self, *args, data: List[ResultInfo] = [], **kwargs) -> None:
```

Comment:
Mutable in default kwargs - could lead to hard to debug problems

`Optional[List[ResultInfo]] = None` could suffice and then `self.result_info = data or []`